src/Datasource/EntityInterface.php

```diff
@@ -149,6 +149,24 @@ public function extract(array $properties, $onlyDirty = false);
      */
     public function dirty($property = null, $isDirty = null);
 
+    /**
+     * Sets the dirty status of a single property.
+     *
+     * @param string $property the field to set or check status for
+     * @param bool $isDirty true means the property was changed, false means
+     * it was not changed
+     * @return $this
+     */
+    public function setDirty($property, $isDirty);
+
+    /**
+     * Checks if the entity is dirty or if a single property of it is dirty.
+     *
+     * @param string $property the field to check the status for
```

Comment:
|null
this is also wrong in master currently :)

Reply:
Fixed in https://github.com/cakephp/cakephp/commit/1c6b605e06b29a411857563924241de6f2a6f37f